cli/test/README.md

```diff
@@ -88,48 +88,48 @@ This document includes CLI commands and the expected CLI output in order to help
 |`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease arg1 arg2 arg3`|`ERROR: set command for Participant builtin discovery config <duration>: 'lease' duration type expects at most 2 arguments and received 3`|`PARTICIPANT_BUILTIN_DISCOVERY_CONFIG_DURATION_USAGE`|
 |`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease infinite`|N/A|N/A|
 |`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease 1`|N/A|N/A|
-|`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease other`|`ERROR: value 'other' does not match any member types of the union`|N/A|
+|`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease other`|`Fast DDS QoS Profiles Manager exception caught: value 'other' does not match any member types of the union`|N/A|
 |`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease arg1 arg2 help`|N/A|`PARTICIPANT_BUILTIN_DISCOVERY_CONFIG_DURATION_USAGE`|
-|`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease 1 500`|**PENDING** (currently `invalid document structure`) [^3]|N/A|
-|`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease infinite 500`|**PENDING** (currently `invalid document structure`) [^3]|N/A|
-|`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease other 500`|**PENDING** (currently `invalid document structure`) [^3]|N/A|
+|`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease 1 500`|**PENDING** (currently `Fast DDS QoS Profiles Manager exception caught: invalid document structure`) [^3]|N/A|
+|`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease infinite 500`|`Fast DDS QoS Profiles Manager exception caught: value 'infinite' does not match any member types of the union`|N/A|
+|`fastddsqosprof file.xml set participant[profile].builtin.discovery_config.duration.lease 1 other`|`Fast DDS QoS Profiles Manager exception caught: value 'other' does not match any member types of the union`|N/A|
```

Comment:
As the argument order has been inverted, in this case the exception returned is the `invalid document structure`. The error you have written is the one that the library will return once the issue is fixed. But I think we should keep this test plan with the current behavior and not the expected one (marking the expected behavior with the footnotes).

Also, this example has just brought to my attention one problem with our CLI usage. Calling to the library sequentially can lead that some of these complex commands succeed and then one of them fails due to some invalid type being passed. But the library is agnostic and does not know that it is being called with the same CLI command. We should decide if we want to support this behavior, if we remove the complex calls from the CLI support or if a refactor is required in the library so the Xerces infrastructure is initialized and used by several calls and then freed by another API call (this would also solve the issue of accessing to the document not yet correctly written).

Please, remember that this suggestion also applies in lease `announcements`.

Reply:
I think we should keep supporting this behaviour, and refactor the API library to support it.